R/incidence.R

```diff
@@ -101,46 +100,27 @@ graph.incidence.dense <- function(incidence, directed, mode, multiple,
 
     n1 <- nrow(incidence)
     n2 <- ncol(incidence)
-    no.edges <- sum(incidence != 0)
-    if (directed && mode == 3) {
-      no.edges <- no.edges * 2
-    }
-    edges <- numeric(2 * no.edges)
-    weight <- numeric(no.edges)
-    ptr <- 1
-    for (i in seq_len(nrow(incidence))) {
-      for (j in seq_len(ncol(incidence))) {
-        if (incidence[i, j] != 0) {
-          if (!directed || mode == 1) {
-            edges[2 * ptr - 1] <- i
-            edges[2 * ptr] <- n1 + j
-            weight[ptr] <- incidence[i, j]
-            ptr <- ptr + 1
-          } else if (mode == 2) {
-            edges[2 * ptr - 1] <- n1 + j
-            edges[2 * ptr] <- i
-            weight[ptr] <- incidence[i, j]
-            ptr <- ptr + 1
-          } else if (mode == 3) {
-            edges[2 * ptr - 1] <- i
-            edges[2 * ptr] <- n1 + j
-            weight[ptr] <- incidence[i, j]
-            ptr <- ptr + 1
-            edges[2 * ptr - 1] <- n1 + j
-            edges[2 * ptr] <- i
-            weight[ptr] <- incidence[i, j]
-            ptr <- ptr + 1
-          }
-        }
-      }
+
+    idx <- which(incidence != 0, arr.ind = TRUE)
+    el <- cbind(idx, incidence[idx])
+
+    el[, 2] <- el[, 2] + n1
```

Comment:
Is this the same as

```r
el <- cbind(idx, incidence[idx] + n1)
```

?

In either case, this needs a bit of explanation in a comment.

Reply:
Yes that is the same. I can refactor a bit and add comments

Reply:
sorry I was wrong here. It is not the same, which makes it obvious that comments are needed...